iOS Client/Reducers/MBAppReducer.swift

```diff
@@ -59,8 +65,27 @@ func devotionReducer(action: Action, state: DevotionState?) -> DevotionState {
     switch action {
     case let action as LoadedDevotions:
         nextState.devotions = action.devotions
+    case let action as SelectedDevotion:
+        nextState.selectedDevotion = action.devotion
+        if case Loaded.loaded(data: var devotions) = nextState.devotions,
+            let index = devotions.index(where: { $0.date == action.devotion.date }) {
+            devotions[index].read = true
+            nextState.devotions = Loaded.loaded(data: devotions)
+        }
+    case let action as DevotionNotification:
+        nextState.selectedDevotion = action.devotion
+        if case Loaded.loaded(data: var devotions) = nextState.devotions,
+            let index = devotions.index(where: { $0.date == action.devotion.date }) {
+            devotions[index].read = true
+            nextState.devotions = Loaded.loaded(data: devotions)
```

Comment:
`lines 69-73` and `76-80` are identical. We need to factor this out into a helper of some sort

Reply:
Or, equivalently, why do we need the DevotionNotification action? Couldn't we just re-use SelectedDevotion and long as you add `nextState.selectedTab = .devotions` to the navigation reducer for SelectedDevotion action?

Reply:
I like the second option I can do that